archey/entries/uptime.py

```diff
@@ -8,41 +17,147 @@ class Uptime(Entry):
+        try:
+            return self._clock_uptime()
+        except RuntimeError:
+            # Probably Python <3.7, or not *NIX
+            pass
+
+        # FUTURE: Windows support could be added here with the `wmi` or `pywin32` module.
+
+        # Fall back to the `uptime` command.
+        return self._parse_uptime_cmd()
+
+    @staticmethod
+    def _proc_file_uptime():
+        """Tries to get uptime using the `/proc/uptime` file"""
+        with open('/proc/uptime') as f_uptime:
+            return timedelta(
+                seconds=float(f_uptime.read().split()[0])
+            )
+
+    @staticmethod
+    def _clock_uptime():
+        """Tries to get uptime using the clocks from the Python `time` module"""
+        # Try: Linux uptime clock, macOS uptime clock, BSD uptime clock.
+        for clock in ('CLOCK_BOOTTIME', 'CLOCK_UPTIME_RAW', 'CLOCK_UPTIME'):
+            try:
+                return timedelta(
+                    seconds=time.clock_gettime(getattr(time, clock))
+                )
+            except AttributeError:
+                pass
+
+        # Probably Python <3.7, or just not one of the above OSes
+        raise RuntimeError
+
+    @staticmethod
+    def _parse_uptime_cmd():
+        """Tries to get uptime by parsing the `uptime` command"""
+        try:
+            uptime_output = check_output('uptime')
+        except FileNotFoundError:
+            # No `uptime` command.
+            # Since `procps` is a dependency (which provides `uptime`) we can just exit here.
+            # Note: We _should_ not got there as `Processes` first check `procps` availability.
+            sys.exit("Please, install first `procps` (or `procps-ng`) on your system.")
+
+        # Unfortunately the output is not designed to be machine-readable...
+        uptime_match = re.search(
+            r"""
+            up\s+?             # match the `up` preceding the uptime, anchors the start of the regex
+            (?:                # non-capture group for days
+               (?P<days>       # 'days' named capture group, captures the days digits
+                  \d+?
+               )
+               \s+?            # match whitespace,
+               days?           # 'day' or 'days',
+               [,\s]+?         # then a comma (if present) followed by more whitespace
+            )?                 # match the days non-capture group 0 or 1 times
+            (?:                # non-capture group for hours & minutes
+               (?:             # non-capture group for just hours
+                  (?P<hours>   # 'hours' named capture group, captures the hours digits
+                     \d+?
+                  )
+                  (?:          # non-capture group for hours:minutes colon or 'hrs' text
+                     :         # i.e. hours followed by either a single colon
+                     |         # OR
+                     \s+?hrs   # one or more whitespace chars non-greedily, followed by 'hrs'
```

Comment:
I took the liberty to remove unecessary `[]` from your pattern, and additional question :

In case there is only `1` hour displayed, does `hrs` get an `s` anyway ?

Reply:
Ah yeah, it looks like it's just `hr` with 1 hour. Here's the [FreeBSD source from the latest release](https://svnweb.freebsd.org/base/release/12.1.0/usr.bin/w/w.c?view=markup) for reference:
```C
513 | if (days > 0)
514 |     sbuf_printf(upbuf, " %d day%s,",
515 |         days, days > 1 ? "s" : "");
516 | if (hrs > 0 && mins > 0)
517 |     sbuf_printf(upbuf, " %2d:%02d,", hrs, mins);
518 | else if (hrs > 0)
519 |     sbuf_printf(upbuf, " %d hr%s,",
520 |         hrs, hrs > 1 ? "s" : "");
521 | else if (mins > 0)
522 |     sbuf_printf(upbuf, " %d min%s,",
523 |         mins, mins > 1 ? "s" : "");
524 | else
525 |     sbuf_printf(upbuf, " %d sec%s,",
526 |         secs, secs > 1 ? "s" : "");
```

Good catch!

Reply:
I've fixed this in 53d5d9f181dd5ace980b8a90c074a49f572275ce 😃